api/Onboard.go

```diff
@@ -0,0 +1,98 @@
+package api
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/operate-first/opfcli/constants"
+	"github.com/operate-first/opfcli/models"
+	"github.com/operate-first/opfcli/utils"
+)
+
+func (api *API) Onboard(path string) error {
+	onboardRequest, err := models.OnboardRequestFromYAMLPath(path)
+	if err != nil {
+		return err
+	}
+
+	if err := api.CreateGroup(
+		onboardRequest.TeamName,
+		onboardRequest.Users,
+		true,
+	); err != nil {
+		return err
+	}
+
+	if err := api.CreateRoleBinding(onboardRequest.TeamName, "admin"); err != nil {
+		return err
+	}
+
+	var tempQuota string
+	var nsPaths []string
+	emptyCustomQuota := models.CustomResourceQuota{}
+
+	for i := range onboardRequest.Namespaces {
+		if onboardRequest.Namespaces[i].CustomQuota != emptyCustomQuota {
+			tempQuota = ""
+		} else if onboardRequest.Namespaces[i].Quota != "" {
+			tempQuota = strings.ToLower(onboardRequest.Namespaces[i].Quota)
+			if err := api.ValidateQuota(tempQuota); err != nil {
+				return err
+			}
+		}
+		if err := api.CreateNamespace(
+			onboardRequest.Namespaces[i].Name,
+			onboardRequest.TeamName,
+			onboardRequest.Namespaces[i].ProjectDisplayName,
+			tempQuota,
+			onboardRequest.Namespaces[i].DisableLimitRange,
+			true,
+		); err != nil {
+			return err
+		}
+		if onboardRequest.Namespaces[i].CustomQuota != emptyCustomQuota {
+			if err := api.CreateCustomResourceQuota(onboardRequest.Namespaces[i].Name, onboardRequest.Namespaces[i].CustomQuota, true); err != nil {
+				return err
+			}
+		}
+
+		nsPaths = append(nsPaths, fmt.Sprintf("../../../../base/core/namespaces/%s", onboardRequest.Namespaces[i].Name))
+	}
+
+	prodOverlayPath := filepath.Join(
+		api.RepoDirectory, api.AppName, constants.ProdOverlayPath, strings.ToLower(onboardRequest.Env), strings.ToLower(onboardRequest.TargetCluster),
+	)
+	prodOverlayKustomizationPath := filepath.Join(prodOverlayPath, "kustomization.yaml")
+
+	exist, err := utils.PathExists(prodOverlayKustomizationPath)
+	if err != nil {
+		return fmt.Errorf("not sure if file exists: %s", prodOverlayKustomizationPath)
+	} else if !exist {
+		return fmt.Errorf("file does not exist: %s", prodOverlayKustomizationPath)
+	}
+
+	err = utils.AddKustomizeResources(prodOverlayPath, nsPaths)
+	if err != nil {
+		return err
+	}
+
+	commonOverlayPath := filepath.Join(api.RepoDirectory, api.AppName, constants.CommonOverlayPath)
```

Comment:
I think we should offer a bit more assertiveness in our logging. Suggestion: 

"Encountered error when trying to verify kustomization in cluster overlay path." 

Same for the other logging statement below.

Reply:
I have made changes to our logging here, take a look and see if you like the changes. Leaving unresolved for now, but resolving the second suggested change on this.